node.go

```diff
@@ -122,9 +122,9 @@ func (node *Node) String() string {
 }
 
 // clone creates a shallow copy of a node with its hash set to nil.
-func (node *Node) clone(version int64) *Node {
+func (node *Node) clone(version int64) (*Node, error) {
 	if node.isLeaf() {
-		panic("Attempt to copy a leaf node")
+		return nil, fmt.Errorf("attempt to copy a leaf node")
```

Comment:
let's create `const` errors and return them.
eg:
```
const ErrCloneLeafNode = fmt.Errorf("attempt to copy a leaf node")
```

Reply:
Added them everywhere I replaced a panic (and the error is not dynamic). More work will be needed on this later on